plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```
@@ -0,0 +1,261 @@
+{
+    private static final Logger log = Logger.get(DeltaLakeTableMetadataScheduler.class);
+
+    private static final String TRINO_LAST_TRANSACTION_VERSION = "trino_last_transaction_version";
+    private static final String TRINO_METADATA_SCHEMA_STRING = "trino_metadata_schema_string";
+    private static final int MAX_FAILED_COUNTS = 10;
+
+    private final DeltaLakeTableOperationsProvider tableOperationsProvider;
+    private final TypeManager typeManager;
+    private final int tableParameterLengthLimit;
+    private final int storeTableMetadataThreads;
+    private final Map<SchemaTableName, TableUpdateInfo> updateInfos = new ConcurrentHashMap<>();
+    private final boolean enabled;
+
+    private ExecutorService executor;
+    private ScheduledExecutorService scheduler;
+    private final AtomicInteger failedCounts = new AtomicInteger();
+
+    @Inject
+    public DeltaLakeTableMetadataScheduler(
+            DeltaLakeTableOperationsProvider tableOperationsProvider,
+            NodeManager nodeManager,
+            TypeManager typeManager,
+            @TableParameterLengthLimit int tableParameterLengthLimit,
+            DeltaLakeConfig config)
+    {
+        this.tableOperationsProvider = requireNonNull(tableOperationsProvider, "tableOperationsProvider is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.tableParameterLengthLimit = tableParameterLengthLimit;
+        this.storeTableMetadataThreads = config.getStoreTableMetadataThreads();
+        requireNonNull(nodeManager, "nodeManager is null");
+        this.enabled = config.isStoreTableMetadataEnabled() && nodeManager.getCurrentNode().isCoordinator();
+    }
+
+    @Managed
+    public boolean isShutdown()
+    {
+        return scheduler.isShutdown();
+    }
+
+    public void putAll(Map<SchemaTableName, TableUpdateInfo> tableParameters)
+    {
+        if (scheduler.isShutdown()) {
+            return;
```

should we throw here? ignoring method calls doesn't seem right

---

yeah, but now the putAll is silently ignored. I'd prefer to explicitly reject the call